stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/textinput/StreamMessageInputView.kt

```diff
@@ -0,0 +1,165 @@
+                ?.let(this::setImageDrawable)
+
+            DrawableCompat.setTintList(
+                drawable,
+                getColorList(
+                    typedArray.getColor(
+                        R.styleable.StreamMessageInputView_lightningButtonIconColor,
+                        ContextCompat.getColor(context, R.color.stream_gray_dark)
+                    ),
+                    typedArray.getColor(
+                        R.styleable.StreamMessageInputView_lightningButtonIconPressedColor,
+                        ContextCompat.getColor(context, R.color.stream_white)
+                    ),
+                    typedArray.getColor(
+                        R.styleable.StreamMessageInputView_lightningButtonIconPressedColor,
+                        ContextCompat.getColor(context, R.color.stream_gray_light)
+                    )
+                )
+            )
+
+            layoutParams.width = typedArray.getDimensionPixelSize(
+                R.styleable.StreamMessageInputView_lightningButtonWidth,
+                context.resources.getDimensionPixelSize(R.dimen.stream_attachment_button_width)
+            )
+
+            layoutParams.height = typedArray.getDimensionPixelSize(
+                R.styleable.StreamMessageInputView_lightningButtonHeight,
+                context.resources.getDimensionPixelSize(R.dimen.stream_attachment_button_height)
+            )
+        }
+    }
+
+    private fun showSendMessageEnabled() {
+        binding.ivSendMessage.setImageResource(R.drawable.stream_ic_filled_up_arrow)
+    }
+
+    private fun hideSendMessageEnabled() {
+        binding.ivSendMessage.setImageResource(R.drawable.stream_ic_filled_right_arrow)
+    }
+
+    private fun configTextInput(typedArray: TypedArray) {
+        TextInputHandler(binding.etMessageTextInput, ::showSendMessageEnabled, ::hideSendMessageEnabled)
+
+        binding.etMessageTextInput.run {
+            setTextColor(
+                typedArray.getColor(
+                    R.styleable.StreamMessageInputView_inputTextColor,
+                    ContextCompat.getColor(context, android.R.color.black)
+                )
+            )
+
+            setHintTextColor(
+                typedArray.getColor(
+                    R.styleable.StreamMessageInputView_inputHintTextColor,
+                    ContextCompat.getColor(context, android.R.color.darker_gray)
+                )
+            )
+
+            textSize =
+                typedArray.getDimensionPixelSize(
+                    R.styleable.StreamMessageInputView_inputTextSize,
+                    context.resources.getDimensionPixelSize(R.dimen.stream_text_size_input)
+                ).toFloat()
+
+            hint = typedArray.getText(R.styleable.StreamMessageInputView_inputHint)
+        }
+    }
+}
+
+private fun getColorList(normalColor: Int, pressedColor: Int, disabledColor: Int) = ColorStateList(
```

Comment:
Was it done intentionally out of the class? Maybe better to move to utils?

Reply:
Makes sense.